pkg/controller/sshd/sshd_controller.go

```diff
@@ -178,9 +169,27 @@ func (r *ReconcileSSHD) Reconcile(request reconcile.Request) (reconcile.Result,
 		}
 	} else {
 		// Request object is being deleted.
-		if controllerutil.ContainsFinalizer(instance, reconcileSSHDFinalizerDNS) && r.route53 != nil {
+		if controllerutil.ContainsFinalizer(instance, reconcileSSHDFinalizerDNS) {
 			r.SetSSHDStatus(instance, "Deleting DNS aliases", cloudingressv1alpha1.SSHDStateFinalizing)
-			if err = r.deleteDNSRecords(); err != nil {
+
+			// fetch the sshd service
+			svc := &corev1.Service{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name}, svc)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			err = r.cloudClient.DeleteSSHDNS(context.TODO(), r.client, instance, svc)
+			switch err {
+			case nil:
+				// all good
+			case err.(*cioerrors.LoadBalancerNotFoundError):
+				// couldn't find the ELB - it's likely still queued for creation
+				r.SetSSHDStatus(instance, "Couldn't reconcile", "AWS ELB isn't ready.")
```

Comment:
nit: This should be worded less cloud-specifically. Maybe: "Load balancer isn't ready yet"